File: model/timezone.go
// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
// See License.txt for license information.

package model

import (
	"encoding/json"
	"io"
)

func TimezonesToJson(timezoneList []string) string {
	b, _ := json.Marshal(timezoneList)
	return string(b)
}

func TimezonesFromJson(data io.Reader) SupportedTimezones {
	var timezones SupportedTimezones
	json.NewDecoder(data).Decode(&timezones)
	return timezones
}

func DefaultUserTimezone() map[string]string {
	defaultTimezone := make(map[string]string)
	defaultTimezone["useAutomaticTimezone"] = "true"
	defaultTimezone["automaticTimezone"] = ""
	defaultTimezone["manualTimezone"] = ""

	return defaultTimezone
}

File: core-plugins/mattermost-plugin-playbooks/e2e-tests/tests/utils/timezone.js
// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import timezones from 'timezones.json';

export function getTimezoneLabel(timezone = '') {
    for (let i = 0; i < timezones.length; i++) {
        const zone = timezones[i];
        if (zone.utc.includes(timezone)) {
            return zone.text;
        }
    }

    return timezone;
}
